index.bs

```diff
@@ -7,36 +7,86 @@ Level: none
 Repository: https://github.com/wintercg/proposal-sockets-api
 Editor: Dominik Picheta, Cloudflare https://cloudflare.com/, dominik@cloudflare.com
+Editor: Ethan Arrowood, Vercel https://vercel.com/, ethan.arrowood@vercel.com
 Abstract: Sockets API for Non-Browser EcmaScript-based runtimes.
 Markup Shorthands: markdown yes
+Markup Shorthands: idl yes
 </pre>
+
 <pre class=link-defaults>
 spec:url; type:interface; text:URL
 spec:html; type:attribute; for:Window; text:navigator
 spec:html; type:method; for:Connect; text:connect
 </pre>
 
-Introduction {#intro}
-=====================
+<h2 id="intro">Introduction</h2>
+
+<div class="non-normative">
 
-*This section is non-normative.*
+<em>This section is non-normative.</em>
 
 This document defines an API for establishing TCP connections in Non-Browser JavaScript runtime
-environments. Existing standard APIs are reused as much as possible, for example ReadableStream
-and WritableStream are used for reading and writing from a socket. Some options are inspired
+environments. Existing standard APIs are reused as much as possible, for example {{ReadableStream}}
+and {{WritableStream}} are used for reading and writing from a [=socket=]. Some options are inspired
 by the existing Node.js `net.Socket` API.
 
-The API described here is already implemented by Cloudflare Workers.
-Workers-specific documentation is available in
-https://developers.cloudflare.com/workers/runtime-apis/tcp-sockets/. This spec shares many
-similarities with that document.
+</div>
+
+<h2 id="concepts">Concepts</h2>
+
+<h3 id="socket-concept-section">Socket</h3>
+
+A <dfn export id="socket-concept">socket</dfn> represents a TCP connection, from which you can read and write data. A socket begins in a <i>connected</i> state (if the socket fails to connect, an error is thrown). While in a <i>connected</i> state, the socket's {{ReadableStream}} and {{WritableStream}} can be read from and written to respectively.
+
+A socket becomes <i>closed</i> when its {{close()}} method is called. A socket configured with `allowHalfOpen: false` will close itself when it receives a FIN or RST packet in its read stream.
+
+<h3 id="connect-concept-section">Connect</h3>
 
-`Socket` interface {#socket-interface}
-======================================
+A socket can be constructed using a <dfn export id="connect-concept">connect</dfn> method defined in a `sockets` module (early implementations may use `vendor:sockets` for the module name), or defined on a [=binding object=].
 
-A `Socket` can be constructed using a {{connect}} method defined in a `sockets` module
-(early implementations may use `vendor:sockets` for the module name),
-or a `connect` method defined on a <a>binding object</a>.
+The connect method is the primary mechanism for creating a [=socket=] instance. It instantiates a socket with a resource identifier and some configuration values. It should syncronously return a socket instance in a <i>connected</i> state (or an error should be thrown).
```

Comment:
```suggestion
The connect method is the primary mechanism for creating a [=socket=] instance. It instantiates a socket with a resource identifier and some configuration values. It should synchronously return a socket instance in a <i>connected</i> state (or an error should be thrown).
```

Reply:
> It should syncronously return a socket instance in a <i>connected</i> state (or an error should be thrown).

Actually it should return a socket instance in a "pending" state. The connection is established asynchronously, but when the socket instance is created that connection might not have yet been created.

Reply:
I can adjust this then

Reply:
Fixed in: https://github.com/wintercg/proposal-sockets-api/pull/10